components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -6,57 +6,15 @@ import Foundation
 import os.log
 import UIKit
 
-open class FxAConfig {
-    // FIXME: these should be lower case.
-    // swiftlint:disable identifier_name
-    public enum Server: String {
-        case Release = "https://accounts.firefox.com"
-        case Stable = "https://stable.dev.lcip.org"
-        case Dev = "https://accounts.stage.mozaws.net"
-    }
-
-    // swiftlint:enable identifier_name
-
-    let contentUrl: String
-    let clientId: String
-    let redirectUri: String
-
-    public init(contentUrl: String, clientId: String, redirectUri: String) {
-        self.contentUrl = contentUrl
-        self.clientId = clientId
-        self.redirectUri = redirectUri
-    }
-
-    public init(withServer server: Server, clientId: String, redirectUri: String) {
-        contentUrl = server.rawValue
-        self.clientId = clientId
-        self.redirectUri = redirectUri
-    }
-
-    public static func release(clientId: String, redirectUri: String) -> FxAConfig {
-        return FxAConfig(withServer: FxAConfig.Server.Release, clientId: clientId, redirectUri: redirectUri)
-    }
-
-    public static func stable(clientId: String, redirectUri: String) -> FxAConfig {
-        return FxAConfig(withServer: FxAConfig.Server.Stable, clientId: clientId, redirectUri: redirectUri)
-    }
-
-    public static func dev(clientId: String, redirectUri: String) -> FxAConfig {
-        return FxAConfig(withServer: FxAConfig.Server.Dev, clientId: clientId, redirectUri: redirectUri)
-    }
-}
-
 public protocol PersistCallback {
     func persist(json: String)
 }
 
-private let queue = DispatchQueue(label: "com.mozilla.firefox-account")
```

Comment:
Removed in favour of a "rust-only queue" (see `fxaRustQueue` and `rustCall` towards the end of this file) that guards access to the rust layer (through `sync` calls).
Asynchronicity is achieved on the higher level by using `DispatchQueue.global().async`

Reply:
@thomcc does the `rustCall` approach I took here look correct to you?